elastic/kibana
# [Remote clusters] Migrate to new ES client

## Summary

See https://github.com/elastic/kibana/issues/73973

## Jest test refactor

Server side jest tests were refactored due to the different API exposed by the client. For each test file the steps taken were:

* Remove the test creator function, e.g. `addRemoteClustersTest` and associated interface `TestOptions`
* Add shared variables and environment setup inside of a `beforeEach`
* Provide a way to create a mocked request object with the ability to swap out the request body
* Then in each test case we mock out the values we expect to receive from the ES endpoint

The intention was to make the tests more standardised with the patterns in our other plugins and to not introduce too much additional "test creator" functionality since this can make the tests harder to maintain.

## How to test

1. Start Kibana and ES with basic license
2. Start a second node with `./.es/8.0.0/bin/elasticsearch -E cluster.name="elasticsearch" -E http.port=9222  -E node.name=node3 -E path.data=/tmp/data -E discovery.type=single-node`
3. Navigate to the remote clusters subsection in the management section
4. Add a new remote cluster, specifying `localhost:9301` as the seed host for the remote cluster, this should be the default publish port for the second node
5. Save the remote cluster, check that it is connected ✅
6. Create a second policy, with dummy values
7. Delete both of the configured clusters
